core/coreapi/swarm.go

```diff
@@ -116,18 +116,16 @@ func (api *SwarmAPI) Peers(context.Context) ([]coreiface.ConnectionInfo, error)
 
 	conns := api.peerHost.Network().Conns()
 
-	var out []coreiface.ConnectionInfo
+	out := make([]coreiface.ConnectionInfo, 0, len(conns))
```

Comment:
And if `len(conns) == 0`...
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

So, in addition to pre-allocating capacity, how about also handling the 0 conns case and getting rid of the append:
```go
    if len(conns) == 0 {
        return nil, nil
    }
    out := make([]coreiface.ConnectionInfo, len(conns))
    for i, c := range conns {
        out[i] = &connInfo{
            ...
        }
    }
    return out, nil
}
```

Reply:
Not sure that's needed, I think it's overhead.

the difference within margin of error.

https://gist.github.com/Lasiar/4b50d7d5dcd2f4aea00f8c79a9c9e0af
![Screenshot_20210716_001951](https://user-images.githubusercontent.com/31652715/125830545-32f08ef4-4fd5-4bbf-b85a-3245d14fc943.png)